### DISCUSSION TITLE
Feature request - iCloud sync

### DISCUSSION THREAD

**Discussion by kkebo** [kkebo/DNSecure#22] [Ideas]
It would be handy to have my profiles and configurations synced between all my devices running DNSecure.

Original: #15

**Comment by ghost** [comment]
I don't think its an good idea. The people want to use the DoT and DoH for enhanced privacy. You know what happens when you put something on iCloud.... Maybe a better solution would be to export the config files via Files app to some external storage and allow to be imported by DNSecure into another device.

**Reply by kkebo** [reply]
@zero0ID Do you understand these protocols? They are just for encryption of DNS requests and responses. It hides from others which domain name you're trying to resolve, i.e. which web server you're trying to access, and of course it's not just the Web.

Thus, I think iCloud sync of configuration is not related to user's privacy. It only contains information about which DNS resolver to use. And it is known on the network path anyway, even if you are using DoT or DoH.

**Comment by ghost** [comment]
Yes, you're right.




8 Jan 2021, 18:50 by notifications@github.com:

**Comment by aniqueta** [comment]
@kkk669 Wondering if this can be put back on the radar? I have multiple devices and propagating changes can be a chore. For folks who are concerned about iCloud data security, there is now end-to-end encryption (E2EE) under Advanced Data Protection (ADP). If this project were to use iCloud Drive to store the settings, the sync would come for "free" (not sure) and if users have ADP turned on, they also get E2EE for "free" (sure). 

Down the line what would also be nice is to import/export settings, so one can manage family devices more easily, for example. Import/export can also be a simpler solution to implement than iCloud Sync (not sure, again if one just needs to store settings in a new place..) 

Thank you for making this app. Much easier than using mobileconfig files. Unfortunately, I'm not an engineer, so cannot help code this up.

**Comment by GabrielBoehme13** [comment]
It's tedious manually copying configurations from one device to recreate them on another. Would really like to have this feature. 🙂

**Reply by kkebo** [reply]
I'm considering the import/export feature in #91.
